src/Client.php

```diff
@@ -91,6 +91,12 @@ public function request($requestOrUri, ?CancellationToken $cancellation = null):
             return $this->retryInterceptor->request($request, $cancellation, $client);
         }
 
+        if ($request->getUri()->getUserInfo() !== '') {
+            throw new \Error('The user information (username:password) component of URIs has been deprecated '
```

Comment:
Not really sure but maybe an `\ErrorException` with `$severity` set to `E_USER_DEPRECATED` might be better?

Reply:
I always felt `ErrorException` was meant to be used in user-defined error handlers. Catching the exception isn't really useful either, so that's why I went with `Error`.

Reply:
This should throw an `HttpException`, as the URL might be user input.